compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -559,9 +559,14 @@ object Parsers {
     def inDefScopeBraces[T](body: => T, rewriteWithColon: Boolean = false): T =
       inBracesOrIndented(body, rewriteWithColon)
 
-    def commaSeparated[T](part: () => T): List[T] =
+    /**
+      * @param readFirst If true, assume we have not read the first `part`. Otherwise,
+      *                  expect that we have (i.e the next thing to expect is a [[COMMA]]).
+      */
+    def commaSeparated[T](part: () => T, readFirst: Boolean = true): List[T] =
```

Comment:
Better to avoid boolean parameters. In this case we can do that by having two methods `commaSeparated` and `commaSeparatedRest`. The second takes the first node as parameter.